src/styles/header.scss

```diff
@@ -0,0 +1,113 @@
+header {
+    position: fixed;
+    z-index: 999;
+    width: 100%;
+    height: 3rem;
+    min-height: 50px;
+    color: #fff;
+    background: rgba(0, 8, 81, 0.8);
+    border: 1px solid transparent;
+    border-radius: 0;
+
+    .input-group {
+      float: right;
+      width: 200px;
+      margin: 0.7rem 0 0;
+
+      @media (max-width: 720px) {
+        width: 150px;
+      }      
+
+      input {
+        height: 25px;
+        font-size: 0.8rem;
+      }
+
+      .bi {
+        color: #fff;
+      }
+
+      .search-link {
+        color: #111111;
+        text-decoration: none;
+      }
+    }
+
+    nav {
+      position: relative;
+      float: right;
+      margin: 0.5rem 1rem 0;
+
+      a {
+        display: inline-block;
+        margin: 0 0.2rem 0;
+
+        &.nav-fav {
+          font-size: 0.8rem;
+          color: #ccc;          
+          text-decoration: none;
+
+          &:hover {
+            color: #fff;
+          }
+        }
+      }
+    }
+
+    i {
+      margin: 0.5rem;
+      font-size: 1.2rem;
+      color: #fff;
+    
+      &:hover {
+        color: #fff;
+      }
+
+      + .star-number {
+        position: absolute;
+        top: 1rem;
+        right: 6.1rem;
+        padding: 0.5rem 0.3rem;
+        font-size: 0.65rem;
+        font-weight: 600;
+        color: #fff;
+        background: red;
+        border-radius: 50%;
+      }     
+      
+      &.bi-film {
+        float: left;
+        margin-left: 1.5rem;
+        font-size: 1.3rem;
```

Comment:
Establishing a structured typography system within our React project will significantly elevate its visual consistency and user experience. By defining a set of typography tokens encompassing font sizes, line heights, and font weights, we can create mixins or functions to consistently apply these values across our components.